python/stretch_diagnostics_tests/power_tests/test_POWER_battery_health.py

```diff
@@ -0,0 +1,180 @@
+import os
+from stretch_body.pimu import Pimu
+import numpy
+import subprocess
+import unittest
+import apt
+from stretch_diagnostics.test_base import TestBase
+from stretch_diagnostics.test_suite import TestSuite
+from stretch_diagnostics.test_runner import TestRunner
+import signal
+
+
+class Test_POWER_battery_health(unittest.TestCase):
+    """
+    Test to check battery health
+    """
+    test = TestBase('test_POWER_battery_health')
+
+    def confirm(self, statement, color):
+        click.secho(statement, fg=color)
+        x = input()
+        if x == 'y':
+            return True
+        if x == 'n':
+            return False
+
+    def apt_check_if_installed(self, pkg):
+        apt_list = apt.Cache()
+        if apt_list[pkg].is_installed is False:
+            if self.confirm("Would you like to install stress y or n", "yellow"):
+                os.system("sudo apt-get install stress")
+                apt_list = apt.Cache()
+        self.assertTrue(apt_list[pkg].is_installed, msg='Stress package not installed')
+
+    def test_stress_package(self):
+        """
+        Check if stress package is installed
+        """
+        apt_list = apt.Cache()
+        self.apt_check_if_installed('stress')
+
+    def test_battery_charged(self):
+        """
+        Check if batteries have been charged and settled
+        """
+        x1 = self.confirm("Have you fully charged the robot y or n", "yellow")
+        self.assertTrue(x1, msg='Not fully charged')
+        
+    def test_charger_unplugged(self):
+        """
+        Check if charger is unplugged and battery voltage has settled
+        """
+        x1 = self.confirm("Is the charger unplugged and the battery voltage has settled for 10 minutes y or n", "yellow")
+        self.assertTrue(x1, msg='Battery voltage not settled')
+
+    def test_applications_closed(self):
+        """
+        Check if no other applications are running in the background
+        """
+        x1 = self.confirm("Have you closed all other applications y or n", "yellow")
+        self.assertTrue(x1, msg='Other applications are still running')
+    def start_stress(self):
+        print('Starting stress')
+        subprocess.Popen("stress -c 4", shell=True)
+        time.sleep(5)
+
+    def kill_stress(self):
+        print('Killing stress')
+        for line in os.popen("ps ax | grep stress | grep -v grep"):
+            fields = line.split()
+            pid = fields[0]
+            os.kill(int(pid), signal.SIGKILL)
+
+    def test_battery_health_test(self):
+        """
+        This tool gathers a baseline voltage reading and a voltage reading under a high load to estimate current state of battery health
+        """
+        global baseline_voltage, load_voltage, baseline_current
+        baseline_reading = {'voltage': [], 'current': []}
+        load_reading = {'voltage': [], 'current': []}
+        sample = 10
+        timeout_cnt = 0
+        timeout = 65
+        p = Pimu()
+        p.startup()
```

Comment:
This line can be removed as p.startup() method is already called within the assertion method.